src/main/java/frc/robot/commands/MotorTestCommand.java

```diff
@@ -0,0 +1,199 @@
+        Intake intake,
+        Drivetrain drivetrain,
+        Conveyor conveyor,
+        Climb climb) {
+
+        this.shooter = shooter;
+        this.intake = intake;
+        this.drivetrain = drivetrain;
+        this.conveyor = conveyor;
+        this.climb = climb;
+
+        intakeIsGood = false;
+        shooterIsGood = false;
+        leftDrivetrainIsGood = false;
+        rightDrivetrainIsGood = false;
+        indexConveyorIsGood = false;
+        beltConveyorIsGood = false;
+        leftClimbIsGood = false;
+        rightClimbIsGood = false;
+        timer = new Timer();
+    }
+
+    @Override
+    public void initialize() {
+        shooterEncoderPos = shooter.getEncoderValue();
+        intakeEncoderPos = intake.getIntakeEncoderPosition();
+        leftDrivetrainEncoderPos = drivetrain.getLeftEncoderPosition();
+        rightDrivetrainEncoderPos = drivetrain.getRightEncoderPosition();
+        indexConveyorEncoderPos = conveyor.getIndexEncoderPosition();
+        beltConveyorEncoderPos = conveyor.getBeltEncoderPosition();
+        leftClimbEncoderPos = climb.getLeftEncoderValue();
+        rightClimbEncoderPos = climb.getRightEncoderValue();
+
+        // TODO - adjust these speeds
+        shooter.setMotorPower(0.3);
+        intake.setIntakeMotorPower(0.3);
+        drivetrain.setBothMotorPower(0.3);
+        conveyor.setIndexMotorPower(0.3);
+        conveyor.setBeltMotorPower(0.3);
+        climb.setMotorPower(0.3);
+        timer.start();
+    }
+
+    @Override
+    public void execute() {
+        if (shooter.getEncoderValue() > shooterEncoderPos + 0.5) {
+            shooter.stop();
+            shooterIsGood = true;
+        }
+
+        if (intake.getIntakeEncoderPosition() > intakeEncoderPos + 0.5) {
+            intake.stop();
+            intakeIsGood = true;
+        }
+
+        if (drivetrain.getLeftEncoderPosition() > leftDrivetrainEncoderPos + 0.5) {
+            drivetrain.setLeftMotorPower(0);
+            leftDrivetrainIsGood = true;
+        }
+
+        if (drivetrain.getRightEncoderPosition() > rightDrivetrainEncoderPos + 0.5) {
+            drivetrain.setRightMotorPower(0);
+            rightDrivetrainIsGood = true;
+        }
+
+        if (conveyor.getIndexEncoderPosition() > indexConveyorEncoderPos + 0.5) {
+            conveyor.setIndexMotorPower(0);
+            indexConveyorIsGood = true;
+        }
+
+        if (conveyor.getBeltEncoderPosition() > beltConveyorEncoderPos + 0.5) {
+            conveyor.setBeltMotorPower(0);
+            beltConveyorIsGood = true;
+        }
+
+        if (climb.getLeftEncoderValue() > leftClimbEncoderPos + 0.5) {
+            climb.setLeftMotorPower(0);
+            leftClimbIsGood = true;
+        }
+
+        if (climb.getRightEncoderValue() > rightClimbEncoderPos + 0.5) {
+            climb.setRightMotorPower(0);
+            rightClimbIsGood = true;
+        }
+
+    }
+
+    @Override
+    public void end(final boolean interrupted) {
+        timer.stop();
```

Comment:
Make sure you stop all the motors when the command ends aswell. This is because the command might get interrupted and the motors will continue to move.

Reply:
That makes sense. I originally had it that way, but deleted it without considering the implications after adding the stop commands to the execute loop.